### PR TITLE
Extra palette tonemap settings

### PR DESCRIPTION
**Pull request by PixelEater2018**
Gives an option for a more neutral colourmap appropriate for Heretic/Hexen and another setting which removes the desaturation from Doom's colourmap. Also fixes a glitch with the palette shader (as seen on rsky2) and improves performance.

### CODE REVIEW

File: src/hwrenderer/postprocessing/hw_postprocess_cvars.cpp [hunk 1]
```diff
@@ -100,5 +100,15 @@ CUSTOM_CVAR(Bool, gl_paltonemap_reverselookup, true, CVAR_ARCHIVE | CVAR_NOINITC
 	screen->UpdatePalette();
 }
 
+CUSTOM_CVAR(Int, gl_paltonemap_colormap, 1, CVAR_ARCHIVE | CVAR_NOINITCALL)
+{
+    screen->UpdatePalette();
```
Comment (raa-eruanna) [inline src/hwrenderer/postprocessing/hw_postprocess_cvars.cpp]: Always do if (screen != nullptr) checks in these CVARs - they can be changed at the command prompt without the screen being active and this will cause a crash.

File: src/hwrenderer/postprocessing/hw_postprocess_cvars.cpp [hunk 2]
```diff
@@ -100,5 +100,15 @@ CUSTOM_CVAR(Bool, gl_paltonemap_reverselookup, true, CVAR_ARCHIVE | CVAR_NOINITC
...
+}
+
+CUSTOM_CVAR(Bool, gl_paltonemap_colormap_nogrey, false, CVAR_ARCHIVE | CVAR_NOINITCALL)
+{
+    screen->UpdatePalette();
```
Comment (raa-eruanna) [inline src/hwrenderer/postprocessing/hw_postprocess_cvars.cpp]: if (screen != nullptr)

**Comment by PixelEater2018** [comment]
Thanks for the reviews. Just to clarify- I need to create multiple new branches and pull them separately? And should this pull request be closed or left open until then?

**Comment by PixelEater2018** [comment]
Here is the before and after the glitch is fixed in the shader:
before
after

**Comment by raa-eruanna** [comment]
I see that you fixed some of my bugs/oversights. I didn't even realize I did those. Thanks for that.

I'm not going to sign off on this, but so far I see no further problems. As for what Alexey asked for, I think we're going to do a squash merge, anyway. What you need to keep in mind for the future, though, is that a) the Github web interface is terrible for editing files, and b) by using it, you created nearly a dozen commits, which makes it harder to bisect bugs because in between these commits one file depends on another and you have effectively created a non-working state per-commit, even if in the final commit everything is working.

Basically, what he's saying, is get a GUI git client. [Here is one for Mac](https://www.sourcetreeapp.com/).

**Comment by alexey-lysiuk** [comment]
> Just to clarify- I need to create multiple new branches and pull them separately?

You should clone your forked repository on local machine.
Create a branch, do all changes, commit them as one commit (or two commits, if it's a new feature and a bug fix for existing one), push this branch.
Go to the main [repository page](https://github.com/coelckers/gzdoom) and GitHub will offer you to create a pull request from recently pushed branch.

>  And should this pull request be closed or left open until then?

You can close it and start from scratch. Another option is to do a force push, but it's more for advanced Git users.

Actual workflow depends on your tools and preferences.
I would create a new branch from the current master, merge this one into it, reset to the base commit from master, commit changes, push to remote. All these steps are easy to do with good Git GUI front-end.
After that, a new pull request can be created via GitHub website.

**Comment by PixelEater2018** [comment]
Thanks guys, a new pull request is up so closing this one now :)
